recipe/session/recipeImplementation.go

```diff
@@ -266,19 +266,16 @@ func makeRecipeImplementation(querier supertokens.Querier, config sessmodels.Typ
 		antiCsrfToken := getAntiCsrfTokenFromHeaders(req)
 		response, err := refreshSessionHelper(recipeImplHandshakeInfo, config, querier, *refreshToken, antiCsrfToken, getRidFromHeader(req) != nil, requestTokenTransferMethod)
 		if err != nil {
+			legacyIdRefreshToken := getCookieValue(req, LEGACY_ID_REFRESH_TOKEN_COOKIE_NAME)
+			unauthorisedErr := errors.UnauthorizedError{}
+			isUnauthorisedErr := defaultErrors.As(err, &unauthorisedErr)
+
 			// This token isn't handled by getToken/setToken to limit the scope of this legacy/migration code
-			if !defaultErrors.As(err, &errors.UnauthorizedError{}) {
-				if getCookieValue(req, LEGACY_ID_REFRESH_TOKEN_COOKIE_NAME) != nil {
+			if legacyIdRefreshToken != nil {
+				if (!isUnauthorisedErr) || (unauthorisedErr.ClearTokens != nil && *unauthorisedErr.ClearTokens) {
```

Comment:
This has been updated/fixed in the node PR.

Reply:
Updated